### PR TITLE
Add support for django 4.2

### PR DESCRIPTION
**Pull request by irtazaakram**
## Description
This is related to ongoing effort to update to Django 4.2 in openedx.
https://github.com/edx/upgrades/issues/381

## Supporting information

Changes:

problem_builder/tests/unit/test_answer_mixin.py
- https://github.com/django/django/blob/f474ba4cb52a41a62da6d6fb3492a95eeb8b3999/django/utils/crypto.py#L51
bit length is required argument for get_random_string in 4.2

pylintrc
> pylint: Command line or configuration file:1: UserWarning: Specifying exception names in the overgeneral-exceptions option without module name is deprecated and support for it will be removed in pylint 3.0. Use fully qualified name (maybe 'builtins.BaseException' ?) instead.

Can we also have a new release when this PR is merged?

Thanks,

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
       matrix:
         os: [ubuntu-20.04]
         python-version: [3.8]
-        toxenv: [py38-django32, integration, quality]
+        toxenv: [py38-django32, py38-django42, integration, quality]
```
Comment (Agrendalath) [inline .github/workflows/ci.yml:31]: We should also run integration tests with Django 4.2.

**Review by Agrendalath — CHANGES_REQUESTED** [review]
Thank you for your contribution, @irtazaakram!

> Can we also have a new release when this PR is merged?

Please see [this comment](https://github.com/openedx/xblock-drag-and-drop-v2/pull/334#issuecomment-1637607077). I explained why we don't need to update the version in such a case. We mistakenly include the tests in the Python package in this repository, but we do not use them anywhere.

**Review by Agrendalath — COMMENTED** [review]
@irtazaakram, the approach looks good, but I'd like to understand where these additional changes (mentioned in the review comments) come from.

File: problem_builder/dashboard.py [hunk 2]
```diff
@@ -156,7 +156,7 @@ def eval_(node):
             elif isinstance(node, ast.Name) and node.id == "x":
                 return x
             elif isinstance(node, ast.BoolOp):  # Boolean operator: either "and" or "or" with two or more values
-                if type(node.op) == ast.And:
+                if isinstance(node.op, ast.And):
```
[the remark this answers is not in the document]
Comment (irtazaakram) [reply problem_builder/dashboard.py:159]: Faced this error for `quality` 

```
quality run-test: commands[1] | pycodestyle problem_builder manage.py setup.py
problem_builder/dashboard.py:159:20: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
ERROR: InvocationError for command /home/runner/work/problem-builder/problem-builder/.tox/quality/bin/pycodestyle problem_builder manage.py setup.py (exited with code 1)
```

File: requirements/base.in [hunk 3]
```diff
@@ -4,3 +4,5 @@
 xblock-utils
 edx-opaque-keys
 Django
+openedx-django-pyfs
```
[the remark this answers is not in the document]
Comment (irtazaakram) [reply requirements/base.in:7]: Faced this error in tests

```
py38-django42 run-test: commands[1] | python manage.py check
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/runner/work/problem-builder/problem-builder/.tox/py38-django42/lib/python3.8/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/home/runner/work/problem-builder/problem-builder/.tox/py38-django42/lib/python3.8/site-packages/django/core/management/__init__.py", line 416, in execute
    django.setup()
  File "/home/runner/work/problem-builder/problem-builder/.tox/py38-django42/lib/python3.8/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/runner/work/problem-builder/problem-builder/.tox/py38-django42/lib/python3.8/site-packages/django/apps/registry.py", line 91, in populate
    app_config = AppConfig.create(entry)
  File "/home/runner/work/problem-builder/problem-builder/.tox/py38-django42/lib/python3.8/site-packages/django/apps/config.py", line 193, in create
    import_module(entry)
  File "/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'djpyfs'
ERROR: InvocationError for command /home/runner/work/problem-builder/problem-builder/.tox/py38-django42/bin/python manage.py check (exited with code 1)

```

File: requirements/base.in [hunk 4]
```diff
@@ -4,3 +4,5 @@
...
+lazy
```
[the remark this answers is not in the document]
Comment (irtazaakram) [reply requirements/base.in]: I've removed these dependencies from `base.in` and moved `XBlock[django]` from `workbench.in` to `test.in`

**Pull request merged**